common/lib/xmodule/xmodule/tests/xml/factories.py

```diff
@@ -134,6 +134,13 @@ def parent(self, _create, extracted, **_):
 
         extracted._xml_node.append(self._xml_node)  # pylint: disable=no-member, protected-access
         extracted.policy.update(self.policy)
+        self._xml_node.attrib.update(  # pylint: disable=no-member, protected-access
+            {
+                key: value
+                for key, value in extracted._xml_node.attrib.iteritems()  # pylint: disable=no-member, protected-access
+                if key in InheritanceMixin.fields.keys()
+            }
+        )
```

Comment:
So rather than updating this code, we need to look into updating the sequence-module tests (at least for self-paced, if not all of them) to not use a test runtime.  Using a test runtime is causing all these issues you are facing with inheritable fields.  

Perhaps the easiest way to get a realistic runtime is to use the standard `ModulestoreTestCase` and get the sequence block from the modulestore.  This may be a violation since the sequence block is within the xmodule library and shouldn't have access to `ModulestoreTestCase` and django.  And so, you may be better off adding your self-paced related tests to a different location for integration testing, making use of django, modulestore, self-paced field overrides, etc.